localstack-core/localstack/services/stepfunctions/stepfunctions_utils.py

```diff
@@ -23,3 +26,42 @@ def _get_result():
         return result
 
     return retry(_get_result, sleep=2, retries=timeout_secs / 2)
+
+
+def get_next_page_token_from_arn(resource_arn: str) -> str:
+    return to_str(base64.b64encode(to_bytes(resource_arn)))
+
+
+_DEFAULT_SFN_MAX_RESULTS: int = 100
+
+
+def validate_and_prepare_pagination_params(
+    max_results: int = 100,
+    next_token: str = None,
+    next_token_length_limit: int = 1024,
+    max_results_upper_limit: int = 1000,
+) -> tuple[int, str]:
+    validation_errors = []
+
+    match max_results:
+        case None | 0:
+            max_results = _DEFAULT_SFN_MAX_RESULTS
+        case int() if max_results > max_results_upper_limit:
+            validation_errors.append(
+                f"Value '{max_results}' at 'maxResults' failed to satisfy constraint: "
+                f"Member must have value less than or equal to {max_results_upper_limit}"
+            )
+
+    match next_token:
+        case str() if len(next_token) > next_token_length_limit:
+            validation_errors.append(
+                f"Value '{next_token}' at 'nextToken' failed to satisfy constraint: "
+                f"Member must have length less than or equal to {next_token_length_limit}"
+            )
+
+    if validation_errors:
+        errors_message = "; ".join(validation_errors)
+        message = f"{len(validation_errors)} validation {'errors' if len(validation_errors) > 1 else 'error'} detected: {errors_message}"
+        raise ValidationException(message)
+
+    return max_results, next_token
```

Comment:
Why do we return `next_token` here? I can't see  this being used

Reply:
You're correct that it's not currently being used. Probably a bit premature but I wanted the function to be flexible enough to potentially return a mutated/edited version of the `next_token` (similar to how the default value of the `max_results` gets set).

Reply:
I understand your point, and conceptually, it does seem appealing. However, I don't think there's enough context provided to this function to produce another next_token.

Additionally, the function is quite crowded. It might be beneficial to separate its responsibilities between normalising and validating. Currently, the function handles validation and returns the max results value if one isn't set. In the future, the next token could be adjusted during the normalisation step instead. What do you think?

Reply:
I coupled this normalisation and validation functionality together since:
- If `max_results` has been normalised, we don't need to check whether it's valid.
- The normalisation step was so trivial (default to `100` if `None` or `0`, I thought including it in the validation step would yield less overhead.

Not sure if this small check necessitates a new function given it would probably just look like:

```python
def normalise_max_results(max_results: int = 100) -> int:
   if not max_results:
      return _DEFAULT_SFN_MAX_RESULTS
   return max_results
```

Ideally, are you suggesting a normalisation function (like above) with

```python
max_results = normalise_max_results(max_result)
assert_pagination_parameters_valid(max_results, next_token) # will raise an Exception if invalid
```

I agree, however, about returning `next_token` being unnecessary, so happy to remove that.

Reply:
I agree the normalisation function would be quite small, however it means the validation function would only validate, like for all the other validation functions in this provider. I believe a separation like this, however small, would work best (also in light of your considerations about having the next_token change in the future).